File: coursera/algorithmic-toolbox/week-3/greedy-algorithms/1_money_change/solution.py
# python3
def get_change(m: int):
    coins, pars = 0, [10, 5, 1]
    for par in pars:
        coins += m // par
        m %= par
    return coins


if __name__ == '__main__':
    print(get_change(int(input())))

File: coursera/algorithmic-toolbox/week-3/greedy-algorithms/1_money_change/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func change(m int64) int64 {
	var coins int64
	pars := []int64{10, 5, 1}
	for _, par := range pars {
		coins += m / par
		m %= par
	}
	return coins
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	_ = scanner.Scan()
	n, _ := strconv.ParseInt(scanner.Text(), 10, 64)
	safe(fmt.Fprintln(os.Stdout, change(n)))
}

func safe(interface{}, error) {}
